src/tools/miri/tests/native-lib/pass/ptr_write_access.rs

```diff
@@ -98,66 +93,66 @@ fn test_init_static_inner() {
     assert_eq!(inner, val);
 }
 
-// Test function that writes a pointer and exposes the alloc of its int argument.
-fn test_expose_int() {
+// Test function that marks an allocation as exposed.
+fn test_exposed() {
     extern "C" {
-        fn expose_int(int_ptr: *const i32, pptr: *mut *const i32);
+        fn ignore_ptr(ptr: *const i32);
     }
 
     let x = 51;
-    let mut ptr = std::ptr::null();
+    let ptr = &raw const x;
+    let p = ptr.addr();
 
-    unsafe { expose_int(&x, &mut ptr) };
-    assert_eq!(unsafe { *ptr }, x);
+    unsafe { ignore_ptr(ptr) };
+    assert_eq!(unsafe { *(p as *const i32) }, x);
 }
 
-// Test function that swaps two pointers and exposes the alloc of an int.
-fn test_swap_ptr() {
+/// Test function that writes a pointer and exposes the alloc of its int argument.
```

Comment:
It's a `int_ptr` argument, isn't it?

Also this doesn't really directly test `expose`. It tests that we can write one pointer to memory through another pointer. `test_swap_ptr` also covers that so I am not sure what the additional point of this test is?

Reply:
Indeed, `test_expose_int` has become redundant: Previously, the idea was that this tested whether the 'initial' allocation in a chain of pointers passed to C (i.e the entry point for our expose-DFS) was itself properly exposed, by dereferencing a wildcard pointer to it. This is now covered by `test_exposed` and `test_return_ptr` (both of which pass a `*const i32`, and we access that `i32` through a wildcard ptr later) - I'll remove this test.